src/main/java/edu/uci/ics/crawler4j/robotstxt/HostDirectives.java

```diff
@@ -44,9 +44,17 @@ public boolean allows(String path) {
     timeLastAccessed = System.currentTimeMillis();
     return !disallows.containsPrefixOf(path) || allows.containsPrefixOf(path);
   }
+  
+  public boolean disallows(String path) {
+    timeLastAccessed = System.currentTimeMillis();
+    return disallows.containsPrefixOf(path) && !allows.containsPrefixOf(path);
+  }
 
   public void addDisallow(String path) {
-    disallows.add(path);
+    if (path.isEmpty())
+        disallows.clear();
```

Comment:
Yes. This makes a commonly found statement work:

User-agent: *
Disallow:

which basically states: grab anything you want. Without this statement, it would add the empty string to the list of disallowed URLs. Since the empty string is a prefix of any string at all, this will disallow all URLs. This is a situation I've encountered in the wild several times during my use of Crawler4J.